mk/cgoflags.sh

```diff
@@ -4,33 +4,32 @@ if [[ -z $MESON_SOURCE_ROOT ]] || [[ -z $MESON_BUILD_ROOT ]] || [[ $# -lt 1 ]];
   echo 'USAGE: ninja -C build cgoflags' >/dev/stderr
   exit 1
 fi
-cd $MESON_SOURCE_ROOT
+cd "$MESON_SOURCE_ROOT"
 
 MK_CGOFLAGS=1
 source mk/cflags.sh
 
 mk_cgoflags() {
-  local PKG=$(realpath --relative-to=. $1)
-  local PKGNAME=$(basename $PKG)
-  local LIBPATH=$(realpath --relative-to=$PKG $MESON_BUILD_ROOT/)
+  local PKG=$(realpath --relative-to=. "$1")
+  local PKGNAME=$(basename "$PKG")
+  local LIBPATH=$(realpath --relative-to="$PKG" "$MESON_BUILD_ROOT/")
 
-  GOFILES=$(find $PKG -maxdepth 1 -name '*.go' -not -name '*_test.go' -not -name 'cgoflags.go')
+  GOFILES=$(find "$PKG" -maxdepth 1 -name '*.go' -not -name '*_test.go' -not -name 'cgoflags.go')
   if [[ -n $GOFILES ]]; then
-    PKGNAME=$(grep -h '^package ' $GOFILES | head -1 | awk '{print $2}')
+    PKGNAME=$(grep -h '^package ' $GOFILES | head -n1 | awk '{print $2}')
```

Comment:
this is kinda ugly and could be replaced by something like

```bash
while read ...; do
  PKGNAME=$(sed ...)
done < <(find ...)
```

but I won't bother...